src/app/[locale]/_components/QrScanner.tsx

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import { useState } from 'react';
+import ScannerLogo from 'assets/ScannerLogo.svg';
+import { Box, CircularProgress, IconButton, useTheme } from '@mui/material';
+import { QrStream } from 'app/[locale]/_components/QrStream';
+import HighlightOffRoundedIcon from '@mui/icons-material/HighlightOffRounded';
+import { useNotificationSpawner } from 'lib/hooks/UseNotificationSpawner';
+import { messages } from 'lib/i18n/localization';
+import { useIntl } from 'react-intl';
+import { LocalizedError } from 'lib/util/LocalizedError';
+
+enum State {
+    Stopped,
+    LoadScanner,
+    ShowVideo,
+    HandleQr,
+}
+
+export function QrScanner(props: { onScan: (scanResult: string) => Promise<void>; size?: number | undefined }) {
+    // Camera and QR on/off logic
+
+    const [state, setState] = useState<State>(State.Stopped);
+
+    const notificationSpawner = useNotificationSpawner();
+    const intl = useIntl();
+
+    const switchToVideoStream = (loadingSuccessful: boolean) => {
+        if (loadingSuccessful) {
+            setState(State.ShowVideo);
+        } else {
+            notificationSpawner.spawn({
+                message: intl.formatMessage(messages.mnestix.qrScanner.errorOnQrScannerOpen),
+                severity: 'error',
+            });
+            setState(State.Stopped);
+        }
+    };
+
+    const handleScan = async (result: string) => {
+        setState(State.HandleQr);
+        try {
+            await props.onScan(result);
+            setState(State.Stopped);
+        } catch (e) {
+            const msg = e instanceof LocalizedError ? e.descriptor : messages.mnestix.qrScanner.defaultCallbackErrorMsg;
+            notificationSpawner.spawn({
+                message: intl.formatMessage(msg),
+                severity: 'error',
+            });
+            setState(State.LoadScanner);
+        }
+    };
+
+    // Scanner logo
+
+    const theme = useTheme();
+    const size = props.size || 250;
+
+    return (
+        <>
+            <Box
+                style={{
+                    cursor: 'pointer',
+                    height: size,
+                    width: size,
+                    margin: 'auto',
+                    position: 'relative',
+                }}
+            >
+                {state === State.Stopped && (
```

Comment:
I used position: 'absolute' on most component, to render them on top of each other. Any improvements on this point?

Reply:
why couldn't you use flex + flexDirection: 'column'? see: https://mui.com/system/flexbox/

Reply:
I need to render the components in front of each other. flexbox wouldnt change that, wouldnt it?